testng-core/src/test/java/test/failedreporter/FailedReporterTest.java

```diff
@@ -67,38 +73,108 @@ public void testExclusionOfPassedConfigsAmidstInheritance() {
     runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>", 1);
   }
 
+  @Test(description = "github-1297")
+  public void testExclusionOfPassedConfigsAmidstInheritanceIndependentOfOrder() {
+    triggerTest(InheritancePassSample.class, InheritanceFailureSample.class);
+    String[] substitutions = new String[] {InheritancePassSample.class.getName()};
+    runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">", 0);
+    substitutions =
+        new String[] {
+          "newTest2", "baseBeforeTest", "baseAfterClass", "baseBeforeClass", "baseBeforeMethod"
+        };
+    runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>", 1);
+  }
+
   @Test(description = "github-1297")
   public void testExclusionOfPassedConfigsInvolvingGroupsAtTestLevel() {
-    triggerTest(GroupsSampleBase.class.getPackage().getName(), true);
-    String[] substitutions = new String[] {GroupsPassSample.class.getName()};
+    triggerTest(GroupsSampleBase.class.getPackage().getName(), true, "run");
+    String[] substitutions =
+        new String[] {GroupsPassSample.class.getName(), GroupsSampleBase.class.getName()};
     runAssertions(mTempDirectory, substitutions, "<class name=\"%s\">", 0);
     substitutions = new String[] {"baseBeforeTest", "baseBeforeClassAlwaysRun", "newTest2"};
     runAssertions(mTempDirectory, substitutions, "<include name=\"%s\"/>", 1);
   }
 
   @Test(description = "github-1297")
   public void testExclusionOfPassedConfigsInvolvingGroupsAtSuiteLevel() {
-    triggerTest(GroupsSampleBase.class.getPackage().getName(), false);
-    String[] substitutions = new String[] {GroupsPassSample.class.getName()};
+    triggerTest(GroupsSampleBase.class.getPackage().getName(), false, "run");
+    String[] substitutions =
+        new String[] {GroupsPassSample.class.getName(), GroupsSampleBase.class.getName()};
```

Comment:
Could you explain this change? Why do you add `GroupsSampleBase`?

Reply:
We're testing `ExclusionOfPassedConfigsInvolvingGroupsAtSuiteLevel` here, and there's no failed config methods in `GroupsSampleBase` class. Also, `GroupsFailureSample` has all config methods inherited from base class. Thus `GroupsSampleBase` class shouldn't be included in rerun report.